include/deal.II/base/type_traits.h

```diff
@@ -0,0 +1,98 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 - 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__type_traits_h
+#define dealii__type_traits_h
+
+
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/block_vector.h>
+#include <deal.II/lac/la_vector.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+
+struct true_type_s {
+static const bool value = true;
+};
+
+/**
+* A true_type similar to C++.11 std::true_type
+* 
+* @author Uwe Köcher, 2017
+*/
+typedef true_type_s true_type;
+
+
+struct false_type_s {
+static const bool value = false;
+};
+
+/**
+* A false_type similar to C++.11 std::true_type
+* 
+* @author Uwe Köcher, 2017
+*/
+typedef false_type_s false_type;
+
+
+/**
+ * A type trait to check if a vector type is non-distributed.
+ * 
+ * @author Uwe Köcher, 2017
+ */
+template<typename T>
+struct is_non_distributed_vector : false_type {};
```

Comment:
Is there a way to enforce that we explicitly specify this trait for every vector type, for example by not deriving the base case from true_type or false_type? This would avoid errors if somebody implements a new vector type and forgets to add this.